### PR TITLE
Update ControllerManager documenation describing some concepts

### PR DESCRIPTION
**Pull request by destogl**
There are some recurring questions recently. Let's start to add them into documentation. This specific one is triggered by [this discourse thread](https://discourse.ros.org/t/ros2-control-controller-restart/24662).

### CODE REVIEW

File: controller_manager/doc/userdoc.rst [hunk 1]
```diff
@@ -90,3 +90,19 @@ There are two scripts to interact with controller manager from launch files:
...
                             Name of the controller manager ROS node
+
+Concepts
+-----------
+
+Restarting all controllers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The simplest way to restart all controllers is by using `switch_controllers` services or CLI and adding all controllers to "start" and "stop" lists.
+Note that not all controllers have to be restarted, e.g., broadcsters.
```
Comment (Briancbn) [inline controller_manager/doc/userdoc.rst:101]:
```suggestion
Note that not all controllers have to be restarted, e.g., broadcasters.
```

File: controller_manager/doc/userdoc.rst [hunk 2]
```diff
@@ -90,3 +90,19 @@ There are two scripts to interact with controller manager from launch files:
...
+
+Restarting hardware
+^^^^^^^^^^^^^^^^^^^^^
+
+If hardware gets restarted then you should go through its lifecycle again. 
+This can be simply achieved by returning ``ERROR`` from ``write`` and ``read`` methods of interface implementation.
+**NOT IMPLEMENTED YET - PLEASE STOP/RESTART ALL CONTROLLERS MANUALLY FOR NOW** The controller manager detects that and stops all the controllers that are commanding that hardware and restarts broadcaters that are listening its states.
```
Comment (Briancbn) [inline controller_manager/doc/userdoc.rst:108]:
```suggestion
**NOT IMPLEMENTED YET - PLEASE STOP/RESTART ALL CONTROLLERS MANUALLY FOR NOW** The controller manager detects that and stops all the controllers that are commanding that hardware and restarts broadcasters that are listening its states.
```

**Review by Briancbn — COMMENTED** [review]
Misspelled broadcaster. Maybe a sphinx spellcheck in the pipeline could help in the future.

**Review by bmagyar — APPROVED** [review]

File: controller_manager/doc/userdoc.rst [hunk 3]
```diff
@@ -90,3 +90,19 @@ There are two scripts to interact with controller manager from launch files:
...
+Restarting all controllers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The simplest way to restart all controllers is by using `switch_controllers` services or CLI and adding all controllers to "start" and "stop" lists.
+Note that not all controllers have to be restarted, e.g., broadcasters.
+
+Restarting hardware
+^^^^^^^^^^^^^^^^^^^^^
+
+If hardware gets restarted then you should go through its lifecycle again.
```
Comment (bmagyar) [inline controller_manager/doc/userdoc.rst:106]:
```suggestion
If hardware gets restarted then you should go through its lifecycle again.
```

File: controller_manager/doc/userdoc.rst [hunk 4]
```diff
@@ -90,3 +90,19 @@ There are two scripts to interact with controller manager from launch files:
...
       -c CONTROLLER_MANAGER, --controller-manager CONTROLLER_MANAGER
                             Name of the controller manager ROS node
+
+Concepts
+-----------
+
+Restarting all controllers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The simplest way to restart all controllers is by using `switch_controllers` services or CLI and adding all controllers to "start" and "stop" lists.
```
Comment (bmagyar) [inline controller_manager/doc/userdoc.rst:100]:
```suggestion
The simplest way to restart all controllers is by using ``switch_controllers`` services or CLI and adding all controllers to ``start`` and ``stop`` lists.
```

**Pull request merged**